low-level/files/opt/seagate/sspl/prerequisites/sspl_dev_deploy.sh

```diff
@@ -0,0 +1,484 @@
+#  Description:   SSPL prerequisites script to setup environment
+#  Purpose    :   Setup yum repos, consul and rabbitmq for SSPL
+#  Usage      :   ./sspl_dev_deploy.sh --help
+# *******************************************************************
+
+set -eE
+
+CORTX_MONITOR_BASE_URL="https://raw.githubusercontent.com/mariyappanp/cortx-monitor/EOS-15396_self_prv"
+SSPL_BASE_DIR="/opt/seagate/cortx/sspl"
+
+LOG_FILE="${LOG_FILE:-/var/log/cortx/sspl/sspl_dev_deploy.log}"
+export LOG_FILE
+
+if [[ ! -e "$LOG_FILE" ]]; then
+    mkdir -p $(dirname "${LOG_FILE}")
+fi
+
+# Config file to store cluster_id & node_id
+CONFIG_FILE="/opt/seagate/cortx/sspl/generated_configs"
+if [[ ! -e "$CONFIG_FILE" ]]; then
+    mkdir -p $(dirname "$CONFIG_FILE")
+    touch $CONFIG_FILE
+fi
+
+echo "INFO: ******** Running $0  ********" 2>&1 | tee -a "${LOG_FILE}"
+echo "INFO: Date: $(date)" 2>&1 | tee -a "${LOG_FILE}"
+
+do_cleanup=false
+install_3rd_party_packages=false
+setup_repo=false
+disable_sub_mgr=false
+skip_bmc=false
+initialize_sspl=false
+TARGET_BUILD=
+RPMS_PATH=
+COMPONENT="SSPL"
+PRODUCT_VERSION="LDR_R2"
+NODE="srvnode-1"
+RMQ_USER="sspluser"
+RMQ_PASSWD=""
+CNTRLR_A="10.0.0.2"
+CNTRLR_B="10.0.0.3"
+CNTRLR_A_PORT="80"
+CNTRLR_B_PORT="80"
+CNTRLR_USER=
+CNTRLR_PASSWD=
+BMC_IP=""
+BMC_USER=""
+BMC_PASSWD=""
+
+
+usage()
+{
+    echo "\
+    SSPL prerequisite script.
+    (Bounded to single node provisioning)
+
+    Usage:
+         $0
+            [-V|--product_version  <LDR_R2>]
+            [-N|--node  <Node name/id>]
+            [-A|--cntrlr_a  <controller A IP>]
+            [-B|--cntrlr_b  <controller B IP>]
+            [-L|--local_rpms_path   <sspl rpms location>]
+            [-T|--target_build_url  <target build url>]
+            [--Ap|--cntrlr_a_port  <controller A Port>]
+            [--Bp|--cntrlr_b_port  <controller B Port>]
+            [-U|--cntrlr_user   <username>]
+            [-P|--cntrlr_pass   <password>]
+            [--i|--bmc_ip     <bmc ipv4 address>]
+            [--u|--bmc_user   <bmc user>]
+            [--p|--bmc_passwd <bmc password>]
+            [--Ru|--rmq_user    <rabbitmq username>]
+            [--Rp|--rmq_passwd  <rabbitmq password>]
+            [--St|--storage_type  <storage type>]
+            [--Sr|--server_type    <server type>]
+            [-D|--disable-sub-mgr]
+            [--standalone_installation]
+            [--initialize_sspl]
+            [--setup_repo]
+            [--cleanup]
+            [-h|--help]
+
+    OPTION:
+    -V      <PRODUCT VERSION>   Product version (Default 'LDR_R2')
+    -N      <NODE NAME>         Default 'srvnode-1'
+    -A      <IP ADDRESS>        IP address of controller A (default '10.0.0.2')
+    -B      <IP ADDRESS>        IP address of controller B (default '10.0.0.3')
+    --Ap    <CNTRLR A PORT>     Controller A port (default '80')
+    --Bp    <CNTRLR A PORT>     Controller B port (default '80')
+    -U      <USER NAME>         Username for controller
```

Comment:
defaults for all, except passwd can be kept

Reply:
Okay.